src/ConnectionTransports/NetworkSocketConnectionTransport.cpp

```diff
@@ -25,335 +25,222 @@ NetworkSocketConnectionTransport::NetworkSocketConnectionTransport(
         {
-            sockaddr_in recvFromAddr{};
-            socklen_t recvFromAddrLen = sizeof(recvFromAddr);
-            int bytesRead = recvfrom(
-                socketHandle,
-                readBuf,
-                sizeof(readBuf),
-                0,
-                reinterpret_cast<sockaddr*>(&recvFromAddr),
-                &recvFromAddrLen);
-            if (bytesRead == -1)
+            buffer.resize(bytesRead);
+
+            // Our peer has closed the connection.
+            // Unless we're a UDP connection, in which case 0 length is a-okay.
+            if (connectionKind != NetworkSocketConnectionKind::Udp)
             {
-                int error = errno;
-                if (error == EINVAL)
-                {
-                    // This means we've closed the socket
-                    break;
-                }
-                else if ((error == EAGAIN) || (error == EWOULDBLOCK))
-                {
-                    // No data was read. Keep trying.
-                }
-                else
-                {
-                    // Unexpected error! Close.
-                    spdlog::error(
-                        "Couldn't read from socket. Error {}: {}",
-                        error,
-                        Util::ErrnoToString(error));
-                    closeConnection();
-                    break;
-                }
+                return Result<ssize_t>::Error("TCP socket closed, read zero bytes");
             }
-            else if (bytesRead == 0)
+            else
             {
-                // Our peer has closed the connection.
-                // Unless we're a UDP connection, in which case 0 length is a-okay.
-                if (connectionKind != NetworkSocketConnectionKind::Udp)
-                {
-                    closeConnection();
-                    return;
-                }
+                return Result<ssize_t>::Success(0);
             }
-            else if (bytesRead > 0)
-            {
-                bool bytesAreFromExpectedAddr = true;
-                if ((connectionKind == NetworkSocketConnectionKind::Udp) && targetAddr.has_value())
-                {
-                    // If we're processing UDP packets, make sure the incoming data is coming
-                    // from the expected address
-                    if (recvFromAddr.sin_addr.s_addr == targetAddr.value().sin_addr.s_addr)
-                    {
-                        // Update our outgoing port to match the source
-                        // TODO: Synchronize this to make sure we don't write before we know the
-                        // correct port.
-                        targetAddr.value().sin_port = recvFromAddr.sin_port;
-                    }
-                    else
-                    {
-                        bytesAreFromExpectedAddr = false;
-                    }
-                }
+        }
+        else if (bytesRead > 0)
+        {
+            buffer.resize(bytesRead);
```

Comment:
Technically I think `reserve` is more semantically accurate, since we don't mean to actually initialize every value here, but I'm not sure it has a real perf impact when it's just bytes.

Reply:
True true, but `resize` is nice because the next line can then use `.size()` to know how big the destination buffer is.